#! /bin/sh /usr/share/dpatch/dpatch-run
## Prevent crash on opj_destroy().dpatch by Robin Cornelius <robin.cornelius@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix a crash path in openjpeg where image->comps is wild and gets used 
## DP: in opj_destroy()

@DPATCH@

--- openjpeg-1.3+dfsg.orig/libopenjpeg/image.c	2007-10-18 13:26:11.000000000 +0100
+++ openjpeg/libopenjpeg/image.c	2009-10-29 09:16:32.578125000 +0000
@@ -28,6 +28,7 @@
 
 opj_image_t* opj_image_create0(void) {
 	opj_image_t *image = (opj_image_t*)opj_malloc(sizeof(opj_image_t));
+	image->comps = NULL;
 	return image;
 }
 
